packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -282,10 +282,25 @@ class MongoshNodeRepl implements EvaluationListener {
 
   async eval(originalEval: asyncRepl.OriginalEvalFunction, input: string, context: any, filename: string): Promise<any> {
     this.lineByLineInput.enableBlockOnNewLine();
-    const shellEvaluator = this.runtimeState().shellEvaluator;
+    const { internalState, repl, shellEvaluator } = this.runtimeState();
+
+    let connectionIssues = false;
     try {
       return await shellEvaluator.customEval(originalEval, input, context, filename);
+    } catch (e) {
+      if (e.name === 'MongoNetworkError' || e.name === 'MongoServerSelectionError') {
+        // we track these errors to prevent getting the default prompt which would
+        // otherwise cause the driver to try and reconnect which will timeout after 30s
+        // - right now we don't have any other way of detecting "disconnection"
+        connectionIssues = true;
+      }
+      throw e;
     } finally {
+      if (connectionIssues) {
```

Comment:
I don’t think we need to care about the prompt for programmatic usage, so something like this might be nice for saving some DB calls when we’re running a script:

```suggestion
      if (connectionIssues || !this.input.isTTY) {
```